common/persistence/persistenceErrorInjectionClients.go

```diff
@@ -0,0 +1,1686 @@
+	return &shardErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewWorkflowExecutionPersistenceErrorInjectionClient creates an error injection client to manage executions
+func NewWorkflowExecutionPersistenceErrorInjectionClient(
+	persistence ExecutionManager,
+	errorRate float64,
+	logger log.Logger,
+) ExecutionManager {
+	return &workflowExecutionErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewTaskPersistenceErrorInjectionClient creates an error injection client to manage tasks
+func NewTaskPersistenceErrorInjectionClient(
+	persistence TaskManager,
+	errorRate float64,
+	logger log.Logger,
+) TaskManager {
+	return &taskErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewHistoryPersistenceErrorInjectionClient creates an error injection HistoryManager client to manage workflow execution history
+func NewHistoryPersistenceErrorInjectionClient(
+	persistence HistoryManager,
+	errorRate float64,
+	logger log.Logger,
+) HistoryManager {
+	return &historyErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewMetadataPersistenceErrorInjectionClient creates an error injection MetadataManager client to manage metadata
+func NewMetadataPersistenceErrorInjectionClient(
+	persistence MetadataManager,
+	errorRate float64,
+	logger log.Logger,
+) MetadataManager {
+	return &metadataErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewVisibilityPersistenceErrorInjectionClient creates an error injection client to manage visibility
+func NewVisibilityPersistenceErrorInjectionClient(
+	persistence VisibilityManager,
+	errorRate float64,
+	logger log.Logger,
+) VisibilityManager {
+	return &visibilityErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+// NewQueuePersistenceErrorInjectionClient creates an error injection client to manage queue
+func NewQueuePersistenceErrorInjectionClient(
+	persistence QueueManager,
+	errorRate float64,
+	logger log.Logger,
+) QueueManager {
+	return &queueErrorInjectionPersistenceClient{
+		persistence: persistence,
+		errorRate:   errorRate,
+		logger:      logger,
+	}
+}
+
+func (p *shardErrorInjectionPersistenceClient) GetName() string {
+	return p.persistence.GetName()
+}
+
+func (p *shardErrorInjectionPersistenceClient) CreateShard(
+	ctx context.Context,
+	request *CreateShardRequest,
+) error {
+	fakeErr := generateFakeError(p.errorRate)
+
+	var persistenceErr error
+	if shouldForwardCallToPersistence(fakeErr) {
+		persistenceErr = p.persistence.CreateShard(ctx, request)
+	}
+
+	if fakeErr != nil {
+		p.logger.Error(msgInjectedFakeErr, tag.Error(fakeErr))
```

Comment:
also include the API name and persistenceErr if not nil?

Reply:
Sure, will update.

Reply:
Updated the log message to include the name of the operation, whether the call is forwarded and the persistenceErr (even if the error is nil, the information is still useful).

All store operations now have pre-defined tags in `log/tag/values.go`